examples/date_check.py

```diff
@@ -0,0 +1,75 @@
+"""
+a pipe-thru program that can filter lines based on panelapp dates
+requires static data, so it partners with parse_panelapp_data.py
+
+bcftools view MY_VCF.vcf.bgz <region> |
+    python date_check.py
+    input/panelapp_137_dates.json
+    2021-01-01
+> newfile.vcf
+
+removes rows containing variants in genes added to
+the specified panel prior to the datetime given
+
+genes without entries in the JSON will default to the threshold
+i.e. will pass into the output
+
+Header lines ignored
+
+Needs to work in python2 inside the container
+the container doesn't have click or argparse installed...
+
+this only works for the Hail formatted VCF export
+"""
+
+import json
+import re
+import sys
+from datetime import datetime
+
+
+ENSG_RE = re.compile(r'vep_gene_id=(?P<gene>ENSG[0-9]*);')
+
+
+def read_pipe():
```

Comment:
```suggestion
def read_pipe() -> Iterator[str]:
```

Reply:
not sure this code is ever going to be used, but making this change